File: app/src/main/java/com/ivanmagda/habito/utils/HabitoListUtils.java
package com.ivanmagda.habito.utils;

import android.support.annotation.NonNull;

import com.ivanmagda.habito.models.ResetFrequency;

import java.util.ArrayList;
import java.util.List;

public final class HabitoListUtils {

    private List<Long> mDates = new ArrayList<>();

    public HabitoListUtils(@NonNull List<Long> dates) {
        this.mDates = new ArrayList<>(dates);
    }

    public List<Long> filteredBy(ResetFrequency.Type type) {
        if (type == ResetFrequency.Type.NEVER) return mDates;

        List<Long> resultList = new ArrayList<>(mDates.size());
        for (Long date : mDates) {
            if (HabitoDateUtils.isDateInType(date, type)) resultList.add(date);
        }

        return resultList;
    }

}

File: app/src/main/java/com/ivanmagda/habito/util/HabitoListUtils.kt
package com.ivanmagda.habito.util

import com.ivanmagda.habito.model.ResetFrequency

class HabitoListUtils(private val dates: List<Long> = ArrayList()) {
    fun filteredBy(type: ResetFrequency.Type): List<Long> {
        if (type == ResetFrequency.Type.NEVER) {
            return dates
        }

        return dates.filter {
            HabitoDateUtils.isDateInType(it, type)
        }
    }
}
